src/main/webapp/view/view.js

```diff
@@ -12,66 +12,75 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import Synchronizer from '../synchronizer.js';
-
-import {intializeDiscussion} from '../view/discussion/discussion.js';
+import DiscussionArea from './discussion/discussion-area.js';
 import TranscriptArea from './transcript/transcript-area.js';
 import Video from './video/video.js';
 
 const ENDPOINT_LECTURE = '/lecture';
 
 const PARAM_ID = 'id';
 
-// TODO: Remove global scope and add to view object.
-window.video = new Video();
+export default class View {
+  #lecture;
+  #lectureId;
 
-/* exported LECTURE_ID */
-window.LECTURE_ID = getLectureId();
+  /** Sets `lectureId`, `lecture` and initializes view page components. */
+  constructor() {
+    this.#lectureId = this.getLectureId();
+    this.getLecture().then((lecture) => {
+      this.#lecture = lecture;
+      this.initialize();
+    });
+  }
 
-// TODO: Remove global scope and link to a View object.
-window.synchronizer = new Synchronizer();
+  /**
+   * Initializes the video player, discussion
+   * and transcript sections for the lecture view page.
+   */
+  async initialize() {
+    this.setLectureName();
 
-/** Sets {@code window.LECTURE} as Lecture for view page. */
-getLecture().then((lecture) => {
-  window.LECTURE = lecture;
-  initialize();
-});
+    // TODO: Make these private once event controller is created.
+    View.video = new Video(this.#lecture);
+    // TODO: Move TranscriptArea initialization outside of initialize()
+    // and replace string parameter with a controller object.
+    View.transcript = new TranscriptArea(this.#lecture, 'event controller');
+    View.discussion = new DiscussionArea(this.#lecture);
 
-/**
- * Initializes the video player, discussion
- * and transcript sections for the lecture view page.
- */
-async function initialize() {
-  setLectureName();
-  window.video.loadVideoApi();
-  intializeDiscussion();
-  // TODO: Move TranscriptArea initialization outside of initialize()
-  // and replace string parameter with a controller object.
-  const transcript = new TranscriptArea('event controller');
-  await transcript.loadTranscript();
-}
+    await View.video.loadVideoApi();
+    await View.transcript.loadTranscript();
+    await View.discussion.initialize();
```

Comment:
why are all of these static?

Reply:
They need to be referenced for the synchronization to work right now